### ISSUE TITLE
STEP: files are not opening due to `<class 'NameError'>: name 'PyExc_FC_AbortIOException' is not defined`

### ISSUE THREAD

**Issue by Sanguinius123** [FreeCAD/FreeCAD#17329]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

step files are not opening.
in the python console the following message is appearing:

```shell
15:29:28  pyException: Traceback (most recent call last):
  File "C:\Program Files\FreeCAD 1.0\bin\Lib\site-packages\freecad\module_io.py", line 12, in OpenInsertObject
    except PyExc_FC_AbortIOException:
           ^^^^^^^^^^^^^^^^^^^^^^^^^
<class 'NameError'>: name 'PyExc_FC_AbortIOException' is not defined
```

### Full version info

OS: Windows 10 build 19045
Word size of FreeCAD: 64-bit
Version: 1.0.0RC2.38806 (Git)
Build type: Release
Branch: (HEAD detached at 1.0rc2)
Hash: https://github.com/FreeCAD/FreeCAD/commit/3d63fc6c2f665a8d5e6468845a419bcac80756c7
Python 3.11.9, Qt 5.15.13, Coin 4.0.3, Vtk 9.2.6, OCC 7.7.2
Locale: Portuguese/Brazil (pt_BR)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/Fusion
Installed mods:

A2plus 0.4.68
Assembly4 0.50.15

### Subproject(s) affected?

None

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by luzpaz** [comment]
Full [About](https://wiki.freecad.org/About) version please, TIA

**Comment by Sanguinius123** [comment]
I have 3 versions of FreeCAD on my PC, 0.21 and the developer version, and in these two versions the step file that I have opens normally, but in the FreeCAD_1.0.0RC2 version that I installed, it gives an error opening this type of file.
I'm going to uninstall them all and download only FreeCAD_1.0.0RC2 to test if there was a conflict between the FreeCAD versions.

**Comment by luzpaz** [comment]
Can you zip the step file and drag/drop it into the github textbox ?

**Comment by Sanguinius123** [comment]
[PRO003_001_VIGA CAIXÃO.zip](https://github.com/user-attachments/files/17468524/PRO003_001_VIGA.CAIXAO.zip)

The version of FreeCAD that I used was FreeCAD_1.0.0RC2-conda-Windows-x86_64-installer-1.
I was able to open this box beam file after saving it in the default FreeCAD file in FreeCAD version 0.21 and then opening it in FreeCAD_1.0.0RC2-conda-Windows-x86_64-installer-1 normally.
I noticed that this is only happening with these files, because it is opening normally in others.
In addition to this file, I am leaving the link to access a Google Drive folder with this file that I attached above and another file.

https://drive.google.com/drive/folders/1P7rYDWZo36bllHZ-HX-tzg76UtB-L1IM?usp=drive_link

**Comment by luzpaz** [comment]
3rd time asking, can you paste your Full [ABOUT](https://wiki.freecad.org/About) info please?!

**Comment by luzpaz** [comment]
Can't reproduce on 
```
OS: Manjaro Linux (KDE/plasma/xcb)
Word size of FreeCAD: 64-bit
Version: 1.1.0dev.39024 (Git) AppImage
Build type: Release
Branch: main
Hash: cf8ad6637351819dfa07d49cfd65d511fe14a084
Python 3.11.9, Qt 5.15.13, Coin 4.0.3, Vtk 9.2.6, OCC 7.7.2
Locale: English/United States (en_US)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/Fusion
```

![Screenshot_20241022_051108](https://github.com/user-attachments/assets/e7bf1839-8286-4222-ab85-67aafb38986d)

**Comment by Sanguinius123** [comment]
I see, it's just that I'm relatively new to FreeCAD and that's why I wasn't understanding (I wasn't clicking on the link).
I'm sorry
this is my version:

OS: Windows 10 build 19045
Word size of FreeCAD: 64-bit
Version: 1.0.0RC2.38806 (Git)
Build type: Release
Branch: (HEAD detached at 1.0rc2)
Hash: 3d63fc6c2f665a8d5e6468845a419bcac80756c7
Python 3.11.9, Qt 5.15.13, Coin 4.0.3, Vtk 9.2.6, OCC 7.7.2
Locale: Portuguese/Brazil (pt_BR)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/Fusion
Installed mods: 
  * A2plus 0.4.68
  * Assembly4 0.50.15


I'm also leaving another file, but it opens normally in this version.

[PRO03_000 - CONJUNTO GERAL PONTE ROLANTE 5T (PARA_DETALHAMENTO).zip](https://github.com/user-attachments/files/17475168/PRO03_000.-.CONJUNTO.GERAL.PONTE.ROLANTE.5T.PARA_DETALHAMENTO.zip)


The old version I use is this one:

[code]
OS: Windows 10 build 19045
Word size of FreeCAD: 64-bit
Version: 0.21.2.33771 (Git)
Build type: Release
Branch: releases/FreeCAD-0-21
Hash: b9bfa5c5507506e4515816414cd27f4851d00489
Python 3.8.10, Qt 5.15.2, Coin 4.0.1, Vtk 8.2.0, OCC 7.6.3
Locale: Portuguese/Brazil (pt_BR)
Installed mods: 
  * A2plus 0.4.68
  * Assembly4 0.50.15
[/code]

**Comment by luzpaz** [comment]
@Sanguinius123 thank you.  
JFYI, 0.21.2 won't be getting any updates. v1.0 will be coming out soon so all attention will be on stabilizing it for release via the RC (Release Candidates). 

Can you test our weekly build (these are 'bleeding-edge' builds of FreeCAD that are released every few days for testing): https://github.com/FreeCAD/FreeCAD-Bundle/releases/tag/weekly-builds  
Please report back.

**Comment by Ankang** [comment]
> @Sanguinius123 thank you. JFYI, 0.21.2 won't be getting any updates. v1.0 will be coming out soon so all attention will be on stabilizing it for release via the RC (Release Candidates). We are currently in RC2.@Sanguinius123谢谢。 JFYI，0.21.2 不会获得任何更新。v1.0 即将发布，因此所有注意力都将集中在稳定它以通过 RC（候选版本）发布上。我们目前处于 RC2 阶段。
> 
> Can you test our weekly build (these are 'bleeding-edge' builds of FreeCAD that are released every few days for testing): https://github.com/FreeCAD/FreeCAD-Bundle/releases/tag/weekly-builds Please report back.您能否测试一下我们的每周版本（这些是 FreeCAD 的“前沿”版本，每隔几天发布一次以进行测试）：https://github.com/FreeCAD/FreeCAD-Bundle/releases/tag/weekly-builds 请回报告。

I found the same problem when I opened the step format file.   After using the weekly-builds, the file can be opened normally. I used FreeCAD_weekly-builds-39024-conda-Windows-x86_64-py311.7z for my test.

**Comment by luzpaz** [comment]
@Ankang please add the full about info for the weekly you tested. TIA

**Comment by luzpaz** [comment]
@chennes could you test on windows ? I can't reproduce on linux (See above: https://github.com/FreeCAD/FreeCAD/issues/17329#issuecomment-2428726077)

**Comment by dmgualb** [comment]
Had the same problem:

12:01:59  pyException: Traceback (most recent call last):
  File "C:\Users\dmgua\Downloads\FreeCAD_1.0.0RC2-conda-Windows-x86_64-py311\bin\Lib\site-packages\freecad\module_io.py", line 12, in OpenInsertObject
    except PyExc_FC_AbortIOException:
           ^^^^^^^^^^^^^^^^^^^^^^^^^
<class 'NameError'>: name 'PyExc_FC_AbortIOException' is not defined

OS: Windows 10 build 19045
Word size of FreeCAD: 64-bit
Version: 1.0.0RC2.38806 (Git)
Build type: Release
Branch: (HEAD detached at 1.0rc2)
Hash: 3d63fc6c2f665a8d5e6468845a419bcac80756c7
Python 3.11.9, Qt 5.15.13, Coin 4.0.3, Vtk 9.2.6, OCC 7.7.2
Locale: Portuguese/Brazil (pt_BR)
Stylesheet/Theme/QtStyle: unset/FreeCAD Classic/Qt default
Installed mods: 
  * A2plus 0.4.68
  * ExplodedAssembly
  * fasteners 0.5.29

[f75_040_g08.zip](https://github.com/user-attachments/files/17589504/f75_040_g08.zip)

**Comment by luzpaz** [comment]
I can't reproduce on 
```
OS: Windows 10 build 19042
Architecture: x86_64
Version: 1.0.0.39109 (Git) Conda
Build type: Release
Branch: (HEAD detached at 1.0.0)
Hash: 2fcc5317fe3aee96ca73475986a577719fc78e20
Python 3.11.10, Qt 5.15.15, Coin 4.0.3, Vtk 9.3.0, OCC 7.8.1
Locale: English/United States (en_US)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/Fusion
```

@Sanguinius123 @dmgualb can you try to reproduce on latest 1.0 please ?

**Comment by adrianinsaval** [comment]
I believe this was fixed by #16847 so I'll close the issue, feel free to reopen if you're still experiencing this in latest versions
